Bluekai parser update

stephanecoquet opened the pull request:
Updated the bluekai parser in order to :
- take into account the format changes in bluekai orange data (new columns)
- make it possible to use the parser with custom parsing rules and schema

valentinlehuger commented on artefact/connectors/bluekai/sdt/line.py:
```diff
@@ -125,43 +119,50 @@ def unrolled_lines(cls, payload):
         return unrolled
 
     @classmethod
+    def fetch_rules(cls, client):
```
for rules and schema the client could be passed in an __init__ function and rules and schema methods could be properties.

valentinlehuger commented on artefact/connectors/bluekai/sdt/__init__.py:
```diff
@@ -0,0 +1,21 @@
+rules_orange = [
```
It would be better to define rules and schemas in file by clients (json maybe or python if it doesn't work with regex)

valentinlehuger commented on artefact/connectors/bluekai/sdt/line.py:
```diff
@@ -125,43 +119,50 @@ def unrolled_lines(cls, payload):
                 }
+                for field in schema:
+                    if field not in ['event_date', 'categories', 'event_type', 'bkuuid']:
+                        output[field] = g(field, default=cls.default_value[field])
```
I think that default_value should be optional (in __init__). We may want an exception if a field is missing.

stephanecoquet commented:
After your review, I : 
- moved schema as an argument for the parse function
- added some verifications of the input schema
- moved all client/task-related data to json files

@valentinlehuger The current approach does not necessitates to instantiate a parser. However, the fact that we are now reading data from files may not be appropriate with this approach. 
What do you think about moving all the client/task data instantiation in an init function (thus changing the way the parser is used), as we quickly discuss ?

Thanks!

valentinlehuger commented on artefact/utils/consume_parser.py:
```diff
@@ -56,11 +65,22 @@ def parse_using_rule(unconsumed, rule):
+                raise ValueError("{} field should be in the line's schema".format(field))
+        for field in input_schema:
+            if field not in full_schema:
+                raise ValueError("{0} field not present in {1} schema".format(field, client))
```
undefined variable `client`

stephanecoquet replied:
fixed

valentinlehuger commented on artefact/utils/consume_parser.py:
```diff
@@ -73,42 +93,43 @@ def timeout_handler(signum, frame):
-        # rules = cls.fetch_rules(client)
-        rules = cls.fetch_rules(client)
-        schema = cls.fetch_schema(client)
+        BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
```
Why variables are suddenly in UPPERCASE ??
